### PR TITLE
Fix redundant not selected announcements for Libre Office Calc

### LINKED ISSUE #8988: LibreOffice Calc: NVDA reports "not selected" for every cell
In NVDA 2018.4beta2, NVDA reports "not selected" for every cell in Calc.

### Steps to reproduce:
1. Open application Calc.
2. Press downArrow to move through cells.

#### Actual behavior:
NVDA reports "not selected" and Cell coordinates.

#### Expected behavior:
NVDA reports Cell coordinates only.

P.S. (If this is the default behavior that NVDA reports "not selected). Could we add an toggle option to turn it on / off?)

#### NVDA Installed/portable/running from source:
Installed.

#### NVDA version:
2018.4beta2

#### Windows version:
6.1.7601 service pack 1

#### Name and version of other software in use when reproducing the issue:

LibreOffice 6.0.7.3

#### Does the issue still occur after restarting your PC?
Yes.

#### Have you tried any other versions of NVDA?
Yes. This issue does not occur in NVDA 2018.3.2.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #8988 

### Summary of the issue:
In LibreOffice Calc, not selected is announced for every cell. This was a regression called by #8898.

### Description of how this pull request fixes the issue:
The focused cell in Calc has the focusable, focused and selectable states, but not the selected state. This causes the unselected announcement. Since it is quite expected that a focused cell is also selected itself, we now add the selected state to selectable and focused.

Whenever I did this, I found out that LibeOffice supports neither accSelection nor IAccessibleTable2, which caused a major issue for LibreOffice. This is now fixed by using a hasattr check.

Even after this we weren't there. Since getSelectedItemsCount returned 0, NVDA still announced the positive selected state. This is now fixed by giving getSelectedItemsCount a default return value of 1, but this fix is open to discussion IMO.

### Testing performed:
Tested using LibreOffice calc.

### Known issues with pull request:
None

### Change log entry:
None needed

### CODE REVIEW

File: source/NVDAObjects/__init__.py [hunk 1]
```diff
@@ -1191,5 +1191,5 @@ def getSelectedItemsCount(self,maxCount=2):
 		Fetches the number of descendants currently selected.
 		For performance, this method will only count up to the given maxCount number, and if there is one more above that, then sys.maxint is returned stating that many items are selected.
 		"""
-		return 0
+		return 1
```
Comment (michaelDCurran) [inline source/NVDAObjects/__init__.py]: I think returning 1 should some how be moved in to specific support for LibreOffice..

**Comment by michaelDCurran** [comment]
I want to be clear on the most important problem here:

In Libre Office Calc, not selected is announced for all cells when 
arrowing around a sheet.

The reason for this is that Libre Office Calc does not class a cell as 
selected unless multiple cells are selected. I.e. moving to a cell with 
arrows and then pressing shift+arrow at least once.

So close to the release of 2018.4 I would only accept a specific fix to 
our support to Libre Office Calc, rather than a change in the logic for 
the newly introduced selection reporting code.

Changing this code globally has the potential of breaking Google Sheets 
in both Chrome and Firefox, not to mention possibly other things. And as 
Google Sheets is currently more accessible than Libre Office Calc, and 
its supported in multiple browsers, I'm afraid, for now, Google Sheets 
wins out here as a priority.

**Comment by leonardder** [comment]
This sounds certainly reasonable. 👍
In that case, I'm tempted to override/remove the selectionContainer altogether from table cells in Libre, which is supposed to create the same behaviour as returning 1 with getSelectedItemsCount on the table. Furthermore, it states more clearly that a Calc table isn't supported as a reliable selection container.

**Review by michaelDCurran — APPROVED** [review]

**Comment by michaelDCurran** [comment]
Did this occur in 2018.3 also?

**Comment by surfer0627** [comment]
No, this case does not occur in 2018.3.2.

**Comment by leonardder** [comment]
This also applies to Google Sheets.

**Pull request merged**